### ISSUE TITLE
Why is the cache folder empty

### ISSUE THREAD

**Issue by MyPrototypeWhat** [vercel/turbo#6753]
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

no

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Linux

### Which canary version will you have in your reproduction?

turbo 1.11.1

### Describe the Bug

<img width="1054" alt="image" src="https://github.com/vercel/turbo/assets/43230886/b6da3517-5635-47eb-886b-f0de3d5ed89c">

`./.turbo-cache` This folder is empty

```
ls -trl ./.turbo-cache ===> total 0

// 
[2023-12-08T08:01:34.075Z] • Packages in scope: @components, components-storybook
[2023-12-08T08:01:34.075Z] • Running build:test in 2 packages
[2023-12-08T08:01:34.075Z] • Remote caching disabled
[2023-12-08T08:01:34.075Z] @components:build:test: cache miss, executing ec9922f811b5f21b
[2023-12-08T08:01:34.075Z] 
[2023-12-08T08:01:34.075Z]  Tasks:    0 successful, 1 total
[2023-12-08T08:01:34.075Z] Cached:    0 cached, 1 total
[2023-12-08T08:01:34.075Z]   Time:    55ms 
[2023-12-08T08:01:34.075Z] 
[2023-12-08T08:01:34.075Z] total 0

```
It's normal on my computer(`macos`), but there was an issue with Jenkins(`Red Hat Enterprise Linux Server release 7.5 (Maipo)`)

### Expected Behavior

cache dir not empty,so that turbo can skip the task

### To Reproduce

`turbo.json`
<img width="562" alt="image" src="https://github.com/vercel/turbo/assets/43230886/24b60e74-542c-4205-8221-0c87a05b9f67">
No matter how many times it is executed, it will always be like this
`ls -trl ./.turbo-cache ===> total 0`

**Comment by MyPrototypeWhat** [comment]
> Can you try running the command with `--no-daemon` and also run with `-vv` on a run with a cache miss where you expect `.turbo-cache` to be populated?您能否尝试运行该命令，并在您期望 `.turbo-cache` 填充的缓存未命中的情况下运行 `--no-daemon` `-vv` 该命令？
> 
> Also can you verify that `turbo` behaves as expected when run with `--go-fallback`?另外，您能否验证 `turbo` 在运行时 `--go-fallback` 是否符合预期？

```
turbo build:test --cache-dir=./.turbo-cache --no-daemon -vv
[2023-12-11T02:05:01.125Z] 
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:00.961+0800 [DEBUG] turborepo_lib::shim: Global turbo version: 1.11.1
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:00.966+0800 [DEBUG] turborepo_lib::shim: Repository Root: /data/jenkinsci/workspace/hy-open-admin-dev_release
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:00.966+0800 [DEBUG] turborepo_lib::shim: No local turbo binary found at: /data/jenkinsci/workspace/hy-open-admin-dev_release/node_modules/turbo-linux-64/bin/turbo
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:00.966+0800 [DEBUG] turborepo_lib::shim: No local turbo binary found at: /data/jenkinsci/workspace/hy-open-admin-dev_release/node_modules/turbo/node_modules/turbo-linux-64/bin/turbo
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:00.966+0800 [DEBUG] turborepo_lib::shim: Local turbo path: /data/jenkinsci/workspace/hy-open-admin-dev_release/node_modules/.pnpm/turbo-linux-64@1.11.1/node_modules/turbo-linux-64/bin/turbo
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:00.966+0800 [DEBUG] turborepo_lib::shim: Local turbo version: 1.11.1
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:00.966+0800 [DEBUG] turborepo_lib::shim: Currently running turbo is local turbo.
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:00.970+0800 [DEBUG] turborepo_lib::commands::run: using the experimental rust codepath
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:00.970+0800 [DEBUG] turborepo_lib::commands::run: configured run struct: Run { base: CommandBase { repo_root: AbsoluteSystemPathBuf("/data/jenkinsci/workspace/hy-open-admin-dev_release"), ui: UI { should_strip_ansi: false }, config: OnceCell(<uninit>), args: Args { version: false, skip_infer: false, no_update_notifier: false, api: None, color: false, cpu_profile: None, cwd: Some("/data/jenkinsci/workspace/hy-open-admin-dev_release"), heap: None, login: None, no_color: false, preflight: false, remote_cache_timeout: None, team: None, token: None, trace: None, verbosity: Verbosity { verbosity: None, v: 2 }, check_for_update: false, test_run: false, run_args: None, command: Some(Run(RunArgs { cache_dir: Some("./.turbo-cache"), cache_workers: 10, concurrency: None, continue_execution: false, dry_run: None, go_fallback: false, single_package: false, filter: [], force: None, framework_inference: true, global_deps: [], graph: None, env_mode: Infer, ignore: [], include_dependencies: false, no_cache: false, no_daemon: true, no_deps: false, output_logs: None, log_order: Auto, only: false, parallel: false, pkg_inference_root: None, profile: None, remote_only: false, remote_cache_read_only: false, scope: [], since: None, summarize: None, log_prefix: Auto, tasks: ["build:test"], pass_through_args: [], experimental_space_id: None })) }, version: "1.11.1" }, processes: ProcessManager(Mutex { data: ProcessManagerInner { is_closing: false, children: [] }, poisoned: false, .. }) }
[2023-12-11T02:05:01.125Z] • Packages in scope: @components, components-storybook
[2023-12-11T02:05:01.125Z] • Running build:test in 2 packages
[2023-12-11T02:05:01.125Z] • Remote caching disabled
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:01.014+0800 [DEBUG] turborepo_lib::run::global_hash: global hash env vars []
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:01.016+0800 [DEBUG] turborepo_lib::run::global_hash: external deps hash: 3a52b581c12511e6
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:01.016+0800 [DEBUG] turborepo_lib::run: global hash: 0119b2b94ca51207
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:01.041+0800 [DEBUG] turborepo_lib::run: running visitor
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:01.041+0800 [DEBUG] turborepo_lib::task_hash: task hash env vars for components-storybook:build:test
[2023-12-11T02:05:01.125Z]  vars: []
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:01.041+0800 [DEBUG] turborepo_lib::task_graph::visitor: task components-storybook#build:test hash is 0627ea411743c2b9
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:01.041+0800 [DEBUG] turborepo_lib::task_hash: task hash env vars for @components:build:test
[2023-12-11T02:05:01.125Z]  vars: []
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:01.041+0800 [DEBUG] turborepo_lib::task_graph::visitor: task @components#build:test hash is ec9922f811b5f21b
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:01.041+0800 [DEBUG] log: Engine visitor dropped callback sender without sending result
[2023-12-11T02:05:01.125Z] @components:build:test: cache miss, executing ec9922f811b5f21b
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:01.041+0800 [DEBUG] turborepo_lib::process: waiting for 0 processes to exit
[2023-12-11T02:05:01.125Z] 
[2023-12-11T02:05:01.125Z]  Tasks:    0 successful, 1 total
[2023-12-11T02:05:01.125Z] Cached:    0 cached, 1 total
[2023-12-11T02:05:01.125Z]   Time:    72ms 
[2023-12-11T02:05:01.125Z] 
[2023-12-11T02:05:01.125Z] 2023-12-11T10:05:01.044+0800 [DEBUG] turborepo_lib::process: waiting for 0 processes to exit
[2023-12-11T02:05:01.125Z] total 0
```

**Comment by MyPrototypeWhat** [comment]
```
turbo build:test --cache-dir='./.turbo-cache' --no-daemon -vv
[2023-12-11T03:16:05.509Z] 
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.306+0800 [DEBUG] turborepo_lib::shim: Global turbo version: 1.11.1-canary.0
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.310+0800 [DEBUG] turborepo_lib::shim: Repository Root: /data/jenkinsci/workspace/hy-open-admin-dev_release
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.310+0800 [DEBUG] turborepo_lib::shim: No local turbo binary found at: /data/jenkinsci/workspace/hy-open-admin-dev_release/node_modules/turbo-linux-64/bin/turbo
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.310+0800 [DEBUG] turborepo_lib::shim: No local turbo binary found at: /data/jenkinsci/workspace/hy-open-admin-dev_release/node_modules/turbo/node_modules/turbo-linux-64/bin/turbo
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.310+0800 [DEBUG] turborepo_lib::shim: Local turbo path: /data/jenkinsci/workspace/hy-open-admin-dev_release/node_modules/.pnpm/turbo-linux-64@1.11.1-canary.0/node_modules/turbo-linux-64/bin/turbo
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.311+0800 [DEBUG] turborepo_lib::shim: Local turbo version: 1.11.1-canary.0
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.311+0800 [DEBUG] turborepo_lib::shim: Currently running turbo is local turbo.
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.314+0800 [DEBUG] turborepo_lib::commands::run: using the experimental rust codepath
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.314+0800 [DEBUG] turborepo_lib::commands::run: configured run struct: Run { base: CommandBase { repo_root: AbsoluteSystemPathBuf("/data/jenkinsci/workspace/hy-open-admin-dev_release"), ui: UI { should_strip_ansi: false }, config: OnceCell(<uninit>), args: Args { version: false, skip_infer: false, no_update_notifier: false, api: None, color: false, cpu_profile: None, cwd: Some("/data/jenkinsci/workspace/hy-open-admin-dev_release"), heap: None, login: None, no_color: false, preflight: false, remote_cache_timeout: None, team: None, token: None, trace: None, verbosity: Verbosity { verbosity: None, v: 2 }, check_for_update: false, test_run: false, run_args: None, command: Some(Run(RunArgs { cache_dir: Some("./.turbo-cache"), cache_workers: 10, concurrency: None, continue_execution: false, dry_run: None, go_fallback: false, single_package: false, filter: [], force: None, framework_inference: true, global_deps: [], graph: None, env_mode: Infer, ignore: [], include_dependencies: false, no_cache: false, no_daemon: true, no_deps: false, output_logs: None, log_order: Auto, only: false, parallel: false, pkg_inference_root: None, profile: None, remote_only: false, remote_cache_read_only: false, scope: [], since: None, summarize: None, log_prefix: Auto, tasks: ["build:test"], pass_through_args: [], experimental_space_id: None })) }, version: "1.11.1-canary.0" }, processes: ProcessManager(Mutex { data: ProcessManagerInner { is_closing: false, children: [] }, poisoned: false, .. }) }
[2023-12-11T03:16:05.509Z] • Packages in scope: @components, components-storybook
[2023-12-11T03:16:05.509Z] • Running build:test in 2 packages
[2023-12-11T03:16:05.509Z] • Remote caching disabled
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.357+0800 [DEBUG] turborepo_lib::run::global_hash: global hash env vars []
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.359+0800 [DEBUG] turborepo_lib::run::global_hash: external deps hash: 6a483703ff193d59
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.359+0800 [DEBUG] turborepo_lib::run: global hash: f1d507e153b75dda
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.381+0800 [DEBUG] turborepo_lib::run: running visitor
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.382+0800 [DEBUG] turborepo_lib::task_hash: task hash env vars for @components:build:test
[2023-12-11T03:16:05.509Z]  vars: []
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.382+0800 [DEBUG] turborepo_lib::task_graph::visitor: task @components#build:test hash is 32fe840a5808c706
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.382+0800 [DEBUG] turborepo_lib::task_hash: task hash env vars for components-storybook:build:test
[2023-12-11T03:16:05.509Z]  vars: []
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.382+0800 [DEBUG] turborepo_lib::task_graph::visitor: task components-storybook#build:test hash is 13ac9ed99522a46c
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.382+0800 [DEBUG] log: Engine visitor dropped callback sender without sending result
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.385+0800 [DEBUG] turborepo_lib::run::cache: log file path: /data/jenkinsci/workspace/hy-open-admin-dev_release/packages/components/.turbo/turbo-build$colon$test.log
[2023-12-11T03:16:05.509Z] @components:build:test: cache hit, replaying logs 32fe840a5808c706
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.385+0800 [DEBUG] turborepo_ui::logs: start replaying logs
[2023-12-11T03:16:05.509Z] @components:build:test: 
[2023-12-11T03:16:05.509Z] @components:build:test: > @components@1.0.0 build:test /data/jenkinsci/workspace/hy-open-admin-dev_release/packages/components
[2023-12-11T03:16:05.509Z] @components:build:test: > cross-env VITE_APP_ENV=test vite build --mode test
[2023-12-11T03:16:05.509Z] @components:build:test: 
[2023-12-11T03:16:05.509Z] @components:build:test: vite v4.5.1 building for test...
[2023-12-11T03:16:05.509Z] @components:build:test: transforming...
[2023-12-11T03:16:05.509Z] @components:build:test: ✓ 1668 modules transformed.
[2023-12-11T03:16:05.509Z] @components:build:test: rendering chunks...
[2023-12-11T03:16:05.509Z] @components:build:test: computing gzip size...
[2023-12-11T03:16:05.509Z] @components:build:test: dist/@components.mjs  117.47 kB │ gzip: 35.98 kB
[2023-12-11T03:16:05.509Z] @components:build:test: dist/@components.umd.js  86.43 kB │ gzip: 32.23 kB
[2023-12-11T03:16:05.509Z] @components:build:test: ✓ built in 3.46s
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.385+0800 [DEBUG] turborepo_ui::logs: finish replaying logs
[2023-12-11T03:16:05.509Z] 
[2023-12-11T03:16:05.509Z]  Tasks:    1 successful, 1 total
[2023-12-11T03:16:05.509Z] Cached:    1 cached, 1 total
[2023-12-11T03:16:05.509Z]   Time:    70ms >>> FULL TURBO
[2023-12-11T03:16:05.509Z] 
[2023-12-11T03:16:05.509Z] 2023-12-11T11:16:05.387+0800 [DEBUG] turborepo_lib::process: waiting for 0 processes to exit
[2023-12-11T03:16:05.509Z] total 0

```

This is a bit inexplicable, but it seems to be working properly now
I just changed the version of pnpm

**Comment by chris-olszewski** [comment]
Do you know which pnpm version was giving you trouble and which version you upgraded to?

**Comment by chris-olszewski** [comment]
Closing as I believe this might have been addressed with #6767
